### DISCUSSION TITLE
How to setup hammerdb for Oracle and run sample workload

### DISCUSSION THREAD

**Discussion by anthonysunny** [TPC-Council/HammerDB#263] [General]
Hi,
 I'm new to hammerDB and Oracle and still exploring the tool. It will be great help to me if there are any how to guides available for Oracle DB workload and running sample workloads using hammerdb CLI on Linux platform.

**Comment by sm-shaw** [comment]
Hi,
If just starting out with HammerDB and Oracle, the best way to ramp-up quickly using the CLI is by first starting with the GUI on a PC or laptop. As the workflow is the same once you understand how to build a schema and run tests using the documentation https://www.hammerdb.com/document.html as a guide you will find it much easier to switch in the CLI by following this guide https://www.hammerdb.com/blog/uncategorized/hammerdb-cli-101/.
I would recommend building Linux test environments in VirtualBox https://www.virtualbox.org/. The examples shown below are from this sort of PC/laptop environment (so intentionally not showing performance). So firstly build your schema and learn how to run a timed test and use the graphical metrics to observe the wait events. The example clearly shows that the bottleneck is the log disk in this case. (log file sync)

![hbdora3](https://user-images.githubusercontent.com/38044085/125602336-bf55e1ce-8403-4cee-8ce6-b855e5bb8470.PNG)

Also learn how to read and understand an AWR report. This example is from the test above, so if you multiply the transactions value by 60 you get the same result as the Oracle TPM. Also look at the top timed events, you can see that this also indicates log file sync as the top wait event.

![hdbora4](https://user-images.githubusercontent.com/38044085/125602343-a79651e8-e5f4-4a85-9b35-25b30012e481.PNG)

Once you are comfortable with this workflow then follow the CLI 101 to translate your GUI actions into command line and firstly run the same tests in your test environment. Once you have done this you will have gained the skills in the quickest way to start running CLI only tests against larger server environments.

**Reply by anthonysunny** [reply]
Hi Shaw,
I'm trying to run the DB workload on Oracle 19c in multi-tenancy in Kubernetes container. Hence trying out via GUI looks difficult. 

Please let me know in case of any pointers for running DB load for Containerised Oracle DB.

**Reply by sm-shaw** [reply]
HammerDB is exactly the same whether the database is running in a container or otherwise. The approach described of first understanding the workflow on a laptop or PC using a databse running under virtualbox is a proven approach to get you from a starting position to building the skills needed quickly. If you prefer to start with the CLI then this guide is here https://www.hammerdb.com/blog/uncategorized/hammerdb-cli-101/.  Otherwise, please refer to the documentation https://www.hammerdb.com/document.html.

**Comment by anthonysunny** [comment]
Thanks Shaw. I agree that cli is also very user friendly as GUI. Do you have any pointers on how to access Database externally when it is running as a container in OpenShift ?

**Reply by sm-shaw** [reply]
Running in containers is out of scope for HammerDB guidance - however a quick search will show a wide number of examples and tools that run HammerDB in this type of environment:
https://www.redhat.com/en/blog/running-microsoft-sql-server-2019-openshift-using-red-hat-openshift-container-storage
https://www.openshift.com/blog/benchmarking-openshift-network-performance-part-1-basics
https://github.com/sagyvolkov/sherlock
https://github.com/cloud-bulldozer/benchmark-operator
https://pypi.org/project/benchmark-runner/

**Reply by anthonysunny** [reply]
Thanks for extended help
